integration-test/src/main/java/com/sequenceiq/it/util/cleanup/CleanupUtil.java

```diff
@@ -172,21 +191,213 @@ public void cleanupCredentials() {
+            }
+            return resourceNames;
+        } catch (JSONException e) {
+            LOG.warn("Cannot get '{}' key, because of: {}", resourceNameType, e.getMessage(), e);
+            return resourceNames;
+        } catch (FileNotFoundException e) {
+            LOG.warn("'{}' file not found, because of: {}", filePath, e.getMessage(), e);
+            return resourceNames;
+        } catch (IOException e) {
+            LOG.warn("Reading '{}' file throws exception: {}", filePath, e.getMessage(), e);
+            return resourceNames;
+        }
+    }
+
+    private void deleteResources(List<String> foundResources, String resourceNameType) {
+        List<Path> fileList = new ArrayList<>();
+        List<String> deletedCredentials = new ArrayList<>();
+        AtomicBoolean e2eCleanupFailed = new AtomicBoolean(false);
+
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(Paths.get(outputDirectory))) {
+            for (Path path : stream) {
+                String fileName = path.getFileName().toString();
+                if (fileName.startsWith("resource_names") && fileName.endsWith(".json")) {
+                    LOG.info("Found resource file: '{}' is going to be added to resource files' list", path.getFileName().toAbsolutePath().normalize());
+                    fileList.add(path);
+                }
+            }
+        } catch (Exception e) {
+            LOG.error("Cannot find resource file at path: '{}', because of: {}", Paths.get(outputDirectory).toAbsolutePath().normalize(), e.getMessage(), e);
+            throw new RuntimeException(String.format("Cannot find resource file at path: '%s', because of: %s",
+                    Paths.get(outputDirectory).toAbsolutePath().normalize(), e.getMessage()));
+        }
+        fileList.forEach(filePath -> {
+            LOG.info("Processing resource file: '{}'", filePath.getFileName());
+            List<String> resourcesName = Optional.ofNullable(getResourcesFromFile(resourceNameType, filePath))
+                    .orElse(List.of(""));
+            resourcesName.forEach(resourceName -> {
+                if (foundResources.stream()
+                        .anyMatch(resourceName::equals)) {
+                    LOG.info("{}:{} will be deleted!", resourceNameType, foundResources.stream()
+                            .filter(resourceName::equals).findAny().orElse(null));
+                    switch (resourceNameType) {
+                        case "distroxName":
+                        case "stackName":
+                            deleteDistrox(getCloudbreakClient(), resourceName);
+                            break;
+                        case "sdxName":
+                            deleteSdx(getSdxClient(), resourceName);
+                            break;
+                        case "credentialName":
+                            deleteCredential(getEnvironmentClient(), resourceName);
+                            e2eCleanupFailed.set(true);
```

Comment:
This is strange. You try to delete credentials, then you immediately say it is failed. Where does this information comes from?

Reply:
Answer in the description:
>If the Cleanup finds one or more credential(s) present from the resource JSON at the given tenant, Cleanup exits with error:
ERROR c.s.it.util.cleanup.CleanupUtil [nolabel] - End To End cleanup have been failed, because of credential(s) '[aws-test-1b8a1afafded4daca3e1cc5e9533f37, yarn-test-54daadca706347818431edc9c6b560]' found left behind! after all the found resources have been deleted/terminated.

Reply:
I see. What about other resources? Don't we expect them to be deleted as well?

Reply:
Credential(s) is(are) the final in the sequence of deleting resources. So we can rest assured if we need to delete any credential from the created ones, that means the related E2E job cleanup was not successful.

We can sure the final deletable resource from the list is the credential. If we need to delete any listed credential from the resources Json file, we need to investigate the related E2E build: 
- What was happened at the Cleanup phase? Is this started at the end of the test execution?
- Why the E2E cleanup cannot fulfil its goal?
In this situation the amount and type (distrox, sdx, environment, credential) of the remained resources are not important for the Cleanup App. The fact: we found deletable resource(s) is the vital information and need to highlight for reviewers.